File: android/src/main/java/dev/les3/capacitor/bundlepath/BundlePathPlugin.java
package dev.les3.capacitor.bundlepath;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "BundlePath")
public class BundlePathPlugin extends Plugin {

    private BundlePath implementation = new BundlePath();

    @PluginMethod
    public void get_path(PluginCall call) {
        String path = call.getString("path");
        JSObject ret = new JSObject();

        ret.put("path", implementation.get_path(path));
        call.resolve(ret);
    }
}

File: ios/Sources/BundlePathPlugin/BundlePathPlugin.swift
import Capacitor
import Foundation

@objc(BundlePathPlugin)
public class BundlePathPlugin: CAPPlugin, CAPBridgedPlugin {
    public let jsName = "BundlePath"
    public let identifier = "BundlePathPlugin"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "get_path", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = BundlePath()

    @objc func get_path(_ call: CAPPluginCall) {
        let value = call.getString("path") ?? ""
        if let path = implementation.get_path(value) {
            call.resolve(["path": path])
        } else {
            call.resolve([:])
        }
    }
}
